scoreboard/controllers.py

```diff
@@ -82,6 +82,31 @@ def register_user(email, nick, password, team_id=None,
                     nick, email, flask.request.remote_addr)
     return user
 
+@utils.require_not_started
+def change_user_team(uid, team_tid, code):
+    """Provide an interface for changing a user's team"""
+    team = models.Team.query.get_or_404(team_tid)
+    user = models.User.query.get_or_404(uid)
+
+    old_team = user.team
+
+    if code.lower() != team.code.lower():
+        raise errors.ValidationError('Invalid team selection or team code')
+
+    if team.tid == user.team_tid:
+        raise errors.ValidationError('Changing to same team')
+
+    app.logger.info('User %s switched to team %s from %s' % (user.nick, team.name, old_team.name))
+    user.team = team
+    user.team_tid = team_tid
+    flask.session['team'] = team_tid
```

Comment:
Couple of ways of preventing this - server side state. Not ideal. Look up cached information each time, also not ideal.

Another way of preventing this is modifying the hmac secret and forcibly logging everyone off at game state change (len(state) + state + len(cookie_secret) + cookie_secret). Not ideal - cookie rotates at the game start, forcing everyone to login again. 

Thought process iteration - new set of state information (game_starts_in_X_minutes) , where the teams can't change X minutes before. Cookie hmac secret is automatically modified at that point in time. Cons: more moving parts, more complexity.

Perhaps there's a way to avoid caching team_tid in sessions - David is currently thinking about this.